packages/space-opera/src/components/poster_controls/poster_controls.ts

```diff
@@ -79,6 +90,16 @@ export class PosterControlsElement extends ConnectedLitElement {
     this.modelViewer.src = '';
     this.modelViewer.src = src;
   }
+
+  onDeletePoster() {
+    dispatchSetPoster(undefined);
+  }
+
+  async onDownloadPoster() {
+    if (!this.modelViewer)
+      return;
+    safeDownloadCallback(await this.modelViewer.toBlob(), 'poster.png', '')();
```

Comment:
Shouldn't this download the poster you already created rather than making a new one?

Reply:
Also, for posters you need the toBlob() option `idealAspect = true`. See tester.ts in modelviewer.dev.

Reply:
both addressed